tests/filecheck/dialects/memref/memref_ops.mlir

```diff
@@ -81,8 +81,8 @@ builtin.module {
 // CHECK-NEXT:     %{{.*}} = memref.alloc(%{{.*}}, %{{.*}}, %{{.*}}) {"alignment" = 0 : i64} : memref<?x?x?xindex>
 // CHECK-NEXT:     %{{.*}} = "memref.alloca"(%{{.*}}) <{"alignment" = 0 : i64, "operandSegmentSizes" = array<i32: 1, 0>}> : (index) -> memref<?xindex>
 // CHECK-NEXT:     %{{.*}} = "memref.alloca"(%{{.*}}, %{{.*}}, %{{.*}}) <{"alignment" = 0 : i64, "operandSegmentSizes" = array<i32: 3, 0>}> : (index, index, index) -> memref<?x?x?xindex>
-// CHECK-NEXT:    %{{.*}} = memref.collapse_shape %{{.*}} [[0 : i64, 1 : i64]] : memref<10x2xindex> into memref<20xindex>
-// CHECK-NEXT:    %{{.*}} = memref.expand_shape %{{.*}} [[0 : i64, 1 : i64]] : memref<20xindex> into memref<2x10xindex>
+// CHECK-NEXT:    %{{.*}} = memref.collapse_shape %{{.*}} [{{#?}}[0 : i64, 1 : i64]] : memref<10x2xindex> into memref<20xindex>
+// CHECK-NEXT:    %{{.*}} = memref.expand_shape %{{.*}} [{{#?}}[0 : i64, 1 : i64]] : memref<20xindex> into memref<2x10xindex>
```

Comment:
Isn't there a way to just escape the `[[`?
Same question for all the `[[` -> `[`

Reply:
Yes, there is! `FileCheck` supports `CHECK{LITERAL}: now I can use {{ and [[ however I want, as this will be interpreted as a literal string!`

Reply:
So you have to choose between escaping those or being able to use any regex and substitution?

Reply:
In combination with `CHECK-SAME` this becomes almost bearable I think. See https://github.com/xdslproject/xdsl/pull/2828/files#diff-19f832aba596f43147f16a174b81a972eff70321b597de54501ea42b60cccb15

Reply:
Yeah okay, that's a solution!